### PR TITLE
Implement dependent based output when they are provided

### LINKED ISSUE #53: Implement dependent based output when they are provided
Implement dependent based output when they are provided in in the input csv

### PR DESCRIPTION
**Pull request by noman404**
Implement dependent based output when they are provided 

For depx = 2, joint household

Added use tax values for states like ['pa', 'nc', 'ca', 'il', 'in', 'ok']

closes #53 #94  #54, #55, #57, #64, #67, #68, #69, #70, #71, #72, #73, #74, #77, #78, #82, #83, #84, #85, #86

### CODE REVIEW

File: policyengine_taxsim/core/input_mapper.py [hunk 1]
```diff
@@ -1,10 +1,95 @@
...
+        for i in range(1, depx + 1):
+            dep_name = f"your {get_ordinal(i)} dependent"
+            household_situation["marital_units"][f"{dep_name}'s marital unit"] = {
+                "members": [dep_name],
+                "marital_unit_id": {str(year): i}
+            }
+    else:
+        household_situation["marital_units"]["your marital unit"]["members"] = (
+            ["you", "your partner"] if mstat == 2 else ["you"]
+        )
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/core/input_mapper.py:54]: We are just adding the dependents to the marital unit right? 
I think we should be adding them as members to the tax_unit, household, family, and spm_unit as well

File: policyengine_taxsim/core/output_mapper.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def export_single_household(taxsim_input, policyengine_situation):
         dict: Dictionary of TAXSIM output variables
     """
     mappings = load_variable_mappings()["policyengine_to_taxsim"]
-
+    print(policyengine_situation)
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/core/output_mapper.py:19]: do we need to keep the print or is it just for internal testing?

**Review by PavelMakarchuk — COMMENTED** [review]
Some questions regarding the implementation of the dependents into the situation

File: policyengine_taxsim/core/input_mapper.py (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_taxsim/core/input_mapper.py:54]: After looking into the issue, it is passed through to all entities

File: tests/test_e2e.py [hunk 3]
```diff
@@ -237,7 +263,351 @@ def test_match_both_output(self):
...
+        full_output_cols = standard_output_cols + [
+            # "tfica"
+            # "v10",  # state_agi
+            # "v13",
+            # "v18",
+            # "v19",
+            # "v26",
+            # "v28",
+            # "v34",
+            # "v45",
```
Comment (PavelMakarchuk) [inline tests/test_e2e.py:589]: why commented?

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_taxsim/core/output_mapper.py (hunk 2 above)
Comment (noman404) [reply policyengine_taxsim/core/output_mapper.py:19]: for internal testing

File: tests/test_e2e.py (hunk 3 above)
Comment (noman404) [reply tests/test_e2e.py:589]: those kept to re-enable again after discussing with tax for the use_tax usage which seems taxsim side does not use

**Pull request merged**